src/test/rustdoc/mixing-doc-comments-and-attrs.rs

```diff
@@ -1,23 +1,15 @@
 #![crate_name = "foo"]
 
 // @has 'foo/struct.S1.html'
-// @count - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]/p' \
-//     1
-// @has - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]/p[1]' \
-//     'Hello world! Goodbye! Hello again!'
+// @snapshot S1_top-doc - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]'
 
 #[doc = "Hello world!\n\n"]
 /// Goodbye!
 #[doc = "  Hello again!\n"]
 pub struct S1;
 
 // @has 'foo/struct.S2.html'
-// @count - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]/p' \
-//     2
-// @has - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]/p[1]' \
-//     'Hello world!'
-// @has - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]/p[2]' \
-//     'Goodbye! Hello again!'
+// @snapshot S2_top-doc - '//details[@class="rustdoc-toggle top-doc"]/div[@class="docblock"]'
```

Comment:
Wouldn't it make more sense to keep the original order for the argument? So first, the file we're checking and then the snapshot file. It would be incoherent with the rest otherwise.

Reply:
I considered that, but then it looks a bit like `S2_top-doc` is a text pattern to match against. Plus, the name of the snapshot is important so you know what file to look at it for the output, so I put it at the front. I'm open to changing it, but I'm not sure if putting the snapshot name at the end would necessarily be clearer.

Reply:
I tihnk coherence is the most important thing. For me it's super strange that only `@snapshot` has this order for the args. If others agree with you, I won't oppose though.

Reply:
Joshua gave a thumbs up on [my previous message](https://github.com/rust-lang/rust/pull/91209#discussion_r761628276), so I think let's keep it at the beginning for now if that's okay.

Reply:
Can always be changed later on in any case so it's fine.